files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/appGroup.js

```diff
@@ -353,35 +358,32 @@ class AppGroup {
     }
 
     getPreferredWidth(actor, forHeight, alloc) {
-        let [iconMinSize, iconNaturalSize] = this.iconBox.get_preferred_width(forHeight);
-        let [labelMinSize, labelNaturalSize] = this.label.get_preferred_width(forHeight);
-        let {iconSpacing} = this.state.settings;
-        // The label text starts in the center of the icon, so we should allocate the space
-        // needed for the icon plus the space needed for(label - icon/2)
-        alloc.min_size = iconNaturalSize + iconSpacing;
-        if (this.state.orientation === St.Side.TOP || this.state.orientation === St.Side.BOTTOM) {
-            let max = this.labelVisible && this.groupState.metaWindows.length > 0 ?
-                labelNaturalSize + iconNaturalSize + iconSpacing : 0;
-            alloc.natural_size = Math.min(iconNaturalSize + Math.max(max, labelNaturalSize), MAX_BUTTON_WIDTH * global.ui_scale);
+        let [iconWidth] = this.iconBox.get_preferred_width(forHeight);
+
+        // alloc.min_size = iconWidth; // future: use same allocation as the rest of the applets
+        alloc.min_size = this.state.trigger('getPanelHeight');
+        if (this.isButtonIconOnly()) {
```

Comment:
This is calling a function to check three conditions that are already in state, in an allocation function that gets called rapidly depending on Clutter's frame rate. If we are going to check state in this function, it would be more optimal if it were a single state property, like `orientation` was.